pilot/pkg/proxy/envoy/v2/ep_filters.go

```diff
@@ -36,7 +38,7 @@ func EndpointsByNetworkFilter(endpoints []endpoint.LocalityLbEndpoints, conn *Xd
 	network, found := conn.modelNode.Metadata[model.NodeMetadataNetwork]
 	if !found {
 		// Couldn't find the sidecar network, using default/local
-		network = ""
```

Comment:
this is not correct. it is not consistent with CDS usage and will likely lead to issues.